src/fobnail-attester.c

```diff
@@ -129,6 +130,90 @@ static void coap_aik_handler(struct coap_resource_t* resource, struct coap_sessi
 
 }
 
+static inline void free_null_ptr(void *ptr)
+{
+    if (ptr != NULL) {
+        free(ptr);
+        ptr = NULL;
+    }
+}
+
+UsefulBuf encode_meta(struct meta_data *meta)
+{
+    UsefulBuf Buffer = NULLUsefulBuf;
+    QCBOREncodeContext ctx;
+    QCBOREncode_Init(&ctx, Buffer);
+
+    QCBOREncode_OpenMap(&ctx);
+        QCBOREncode_AddUInt64ToMap(&ctx, "version", meta->header_version);
+        QCBOREncode_OpenArrayInMap(&ctx, "mac");
+        for (int i = 0; i < 6; i++)
```

Comment:
@0xDen Please use QCBOREncode_AddBytesToMap instead

```suggestion
UsefulBufC mac = { meta->mac, sizeof(meta->mac) };
QCBOREncode_AddBytesToMap(&ctx, "mac", mac);
```

Reply:
Fixed.